src/main/scala/dev/kamu/cli/commands/AddCommand.scala

```diff
@@ -3,22 +3,32 @@ package dev.kamu.cli.commands
 import dev.kamu.cli.{
   AlreadyExistsException,
   MetadataRepository,
-  MissingReferenceException
+  MissingReferenceException,
+  SchemaNotSupportedException
 }
 import org.apache.hadoop.fs.{FileSystem, Path}
 import org.apache.log4j.LogManager
+import java.net.{URI, URISyntaxException}
 
 class AddCommand(
   fileSystem: FileSystem,
   metadataRepository: MetadataRepository,
-  manifests: Seq[Path]
+  manifests: Seq[String]
 ) extends Command {
   private val logger = LogManager.getLogger(getClass.getName)
 
   def run(): Unit = {
-    val sources = manifests.map(manifestPath => {
-      logger.debug(s"Loading dataset from: $manifestPath")
-      metadataRepository.loadDatasetFromFile(manifestPath)
+    val sources = manifests.map(manifestStr => {
+      logger.debug(s"Loading dataset from: $manifestStr")
+      try {
+        val manifestURI = new URI(manifestStr);
+        if (null == manifestURI.getScheme)
+          metadataRepository.loadDatasetFromFile(new Path(manifestStr))
+        else metadataRepository.loadDatasetFromURI(manifestURI)
+      } catch {
+        case _: URISyntaxException =>
```

Comment:
I think this complexity can be entirely avoided if we assume that upper level logic should do the necessary escaping/encoding of file paths that contain non-URL-friendly characters, e.g. if there is a space it will be converted to `%20`. Then this class can then accept a `Seq[URI]` as a parameter and avoid this exception handling.

Reply:
I am not sure how and where to add that upper level logic, since the argument parsing is done by another library, which seem not to handle URI. This is what I get if I remove that logic:
```
kamu add ~/.local/share/wineprefixes/steam/dosdevices/c\:/Program\ Files\ \(x86\)/ca.vancouver.data.local-area-boundaries.yaml 
Exception in thread "main" java.net.URISyntaxException: Illegal character in path at index 64: /home/nick/.local/share/wineprefixes/steam/dosdevices/c:/Program Files (x86)/ca.vancouver.data.local-area-boundaries.yaml
```
If I keep it it works just as expected:
```
kamu add ~/.local/share/wineprefixes/steam/dosdevices/c\:/Program\ Files\ \(x86\)/ca.vancouver.data.local-area-boundaries.yaml 
[INFO] Added 1 datasets
```

Reply:
Looks like the previous implementation also fails with the above example. There is a bug in either the library or scala or both.
```
kamu add ~/.local/share/wineprefixes/steam/dosdevices/c\:/Program\ Files\ \(x86\)/ca.vancouver.data.local-area-boundaries.yaml 
[ERROR] Excess arguments provided: '/home/nick/.local/share/wineprefixes/steam/dosdevices/c:/Program Files (x86)/ca.vancouver.data.local-area-boundaries.yaml'
```

Reply:
Oh, I've been missing the converters, which were producing the issue. Should be fixed in the next commit.